### PR TITLE
Zerodha frontend -> integrations

### PR DESCRIPTION
**Pull request by nickmccullum**

### CODE REVIEW

File: src/components/RebalanceWidget/index.tsx [hunk 1]
```diff
@@ -97,6 +97,30 @@ const RebalanceWidget = ({
       });
   };
 
+  const calculateZerodhaTrades = () => {
+    const t = trades.trades;
+    const zerodhaTradeBasket: any = [];
```
Comment (brendanwood) [inline src/components/RebalanceWidget/index.tsx:102]: there's a cleaner way to do this where you're not modifying `zerodhaTradeBasket` directly inside your forEach loop. should instead do something like this:
```
const zerodhaTradeBasket: TradeBasketType = trades.trades.map(trade => {
    return {
        variety: 'regular',
        tradingsymbol: trade.universal_symbol.description,
        ...blah blah....
    };
})
```
Having an explicit type is probably helpful here to make sure we're sending in the right object in all cases.

File: src/components/RebalanceWidget/index.tsx [hunk 2]
```diff
@@ -166,6 +190,36 @@ const RebalanceWidget = ({
     </div>
   );
 
+  const zerodhaAccounts = groupAccounts.filter(
```
Comment (brendanwood) [inline src/components/RebalanceWidget/index.tsx:193]: I don't think this is the proper way to filter by brokerage. You should use the `account.brokerage_authorization` field to lookup the relevant brokerage auth from the authorizations redux field, and then check the associated brokerage slug there.

File: src/components/RebalanceWidget/index.tsx [hunk 3]
```diff
@@ -166,6 +190,36 @@ const RebalanceWidget = ({
...
+    (acc: any) => acc.meta.institution_name === 'Zerodha',
+  );
+
+  if (zerodhaAccounts) {
+    orderValidation = (
```
Comment (brendanwood) [inline src/components/RebalanceWidget/index.tsx:198]: What happens here if more than one Zerodha account is connected?

File: src/pages/AuthorizationPage.tsx [hunk 4]
```diff
@@ -183,6 +184,22 @@ const AuthorizationPage = ({ onboarding }: Props) => {
         </P>
       ),
     },
+    {
+      id: 'zerodha',
+      name: 'Zerodha',
+      connect: () => {
+        startConnection('Zerodha', 'trade');
+      },
+      openURL: 'https://kite.trade/connect/login?api_key=pnriechdkzx5ipvq&v=3',
```
Comment (brendanwood) [inline src/pages/AuthorizationPage.tsx:193]: need to add a couple fields here (the structure changed recently). specifically, `defaultConnectionType: 'true',` and `confirmPrompt: null,`

**Review by brendanwood — CHANGES_REQUESTED** [review]

File: src/components/RebalanceWidget/index.tsx (hunk 3 above)
Comment (nickmccullum) [reply src/components/RebalanceWidget/index.tsx:198]: Right now it's not possible for one user to own multiple Zerodha accounts: https://support.zerodha.com/category/your-zerodha-account/your-profile/articles/multiple-trading-accounts#:~:text=No%2C%20you%20can%20only%20open,name%20of%20your%20family%20members.

Previously we had talked about launching assuming people would only pair one account per portfolio group, and adjusting later if that isn't the case

**Pull request closed without merging**